config/management/operational/src/validator_config.rs

```diff
@@ -101,3 +122,86 @@ impl SetValidatorConfig {
         Ok(TransactionContext::new(self.owner_account, sequence_number))
     }
 }
+
+#[derive(Debug, StructOpt)]
+pub struct RotateValidatorNetworkKey {
+    #[structopt(long, help = "Chain Id representing the chain (one byte)")]
+    chain_id: u8,
+    #[structopt(long, help = "JSON-RPC Endpoint (e.g. http://localhost:8080")]
+    host: String,
+    #[structopt(long, help = "Owner account")]
+    owner_account: AccountAddress,
+    #[structopt(flatten)]
+    validator_backend: ValidatorBackend,
+}
+
+impl RotateValidatorNetworkKey {
+    pub fn execute(self) -> Result<(TransactionContext, x22519::PublicKey), Error> {
+        let idx_addr = 0;
+        let storage_name = self.validator_backend.name();
+
+        // Get current sequence number
+        let client = JsonRpcClientWrapper::new(self.host);
+
+        // TODO: This should be in a single call for both validator config and sequence number
+        let sequence_number = client.get_sequence_number(self.owner_account)?;
+
+        let mut validator_config = client.get_validator_config(owner_account)?;
+        let fullnode_network_key = validator_config
+            .full_node_network_identity_public_key
+            .clone();
+        let mut validator_address = decode_validator_address(
+            validator_config.validator_network_address,
+            &self.owner_account,
+            &TEST_SHARED_VAL_NETADDR_KEY,
+            addr_idx,
+        )?;
+
+        let mut storage = StorageWrapper::new(
+            self.validator_backend.name(),
+            &self.validator_backend.validator_backend,
+        )?;
+
+        // Rotate key
+        let new_validator_network_key = storage.rotate_key(VALIDATOR_NETWORK_KEY)?;
+        let new_validator_network_key = StorageWrapper::to_x25519(new_validator_network_key)?;
+        validator_address.rotate_noise_public_key(
+            &validator_config.validator_network_identity_public_key,
+            &new_validator_network_key,
+        );
+
+        let raw_enc_validator_address = encode_validator_address(
```

Comment:
yeah, I think there's a clear pattern here :), this would also use the common ValidatorConfig + structopt flatten